finality-provider/service/fp_manager.go

```diff
@@ -31,15 +30,15 @@ func (ce *CriticalError) Error() string {
 	return fmt.Sprintf("critical err on finality-provider %s: %s", ce.fpBtcPk.MarshalHex(), ce.err.Error())
 }
 
+// FinalityProviderManager is responsible to initiate and start the given finality
+// provider instance, monitor its running status
 type FinalityProviderManager struct {
```

Comment:
Have we considered removing one of FP manager or FP instance completely? The manager-instance separation is mostly due to the multi FP feature but now we no longer need it

Reply:
I agree we probably don't need the manager and the goroutines in it can be moved to the instance itself. Will do it in a separate PR as it will touch more things